src/qml/DeviceLock.qml

```diff
@@ -0,0 +1,62 @@
+import QtQuick 2.1
+import QtQuick.Controls 1.0
+import QtQuick.Controls.Nemo 1.0
+import QtQuick.Controls.Styles.Nemo 1.0
+import QtQuick.Layouts 1.0
+
+Item {
+    id: root
+    anchors.top: clock.bottom
+    anchors.bottom: parent.bottom
+    anchors.left: parent.left
+    anchors.right: parent.right
+    ColumnLayout {
+        anchors.fill: parent
+        spacing: 40
+        TextField {
+            id: lockCodeField
+            readOnly: true
+            echoMode: TextInput.PasswordEchoOnEdit
+        }
+
+        GridLayout {
+            height: parent.height
+            width: parent.width
+            columns: 3
+            Repeater {
+                model: [0,1,2,3,4,5,6,7,8,9,10,11]
```

Comment:
Yeah, I don't really know how to make a mixed model like that, do you have any suggestions?